### PR TITLE
Customizing_registration_page_in_devise first time

### PR DESCRIPTION
**Pull request by mdmahmudurrahman**

### CODE REVIEW

**Comment by shakil-fg** [comment]
ok

**Comment by noor-framgia** [comment]
ok

File: app/controllers/registrations_controller.rb [hunk 1]
```diff
@@ -6,4 +16,14 @@ def sign_up_params
     def account_update_params
       params.require(:user).permit(:first_name, :last_name, :email, :password, :password_confirmation, :current_password)
     end
+
+    def check_supervisor
+      if user_signed_in?
+        if current_user.supervisor?
+         true
+        else
+         redirect_to root_url
```
Comment (manazirahsan) [inline app/controllers/registrations_controller.rb]: indentation

Comment (manazirahsan) [inline app/controllers/registrations_controller.rb]: indentation

**Comment by manazirahsan** [comment]
okey

File: app/controllers/registrations_controller.rb [hunk 2]
```diff
@@ -1,4 +1,14 @@
...
+    super
+  end
+
+  def create
+    super
+  end
+
   private
     def sign_up_params
       params.require(:user).permit(:name, :email, :password, :password_confirmation, :role)
```
Comment (MuhammadTamzid) [inline app/controllers/registrations_controller.rb]: remove ()

File: app/controllers/registrations_controller.rb [hunk 3]
```diff
@@ -6,4 +16,14 @@ def sign_up_params
     def account_update_params
       params.require(:user).permit(:first_name, :last_name, :email, :password, :password_confirmation, :current_password)
```
Comment (MuhammadTamzid) [inline app/controllers/registrations_controller.rb]: remove ()

File: app/controllers/registrations_controller.rb [hunk 4]
```diff
@@ -6,4 +16,14 @@ def sign_up_params
...
     end
+
+    def check_supervisor
+      if user_signed_in?
```
Comment (dieunb) [inline app/controllers/registrations_controller.rb]: `+      redirect_to root_url if user_signed_in? && !current_user.supervisor?
`

**Comment by noor-framgia** [comment]
ok

**Comment by manazirahsan** [comment]
Okey

**Comment by manazirahsan** [comment]
okey

File: app/controllers/registrations_controller.rb [hunk 5]
```diff
@@ -1,9 +1,31 @@
 class RegistrationsController < Devise::RegistrationsController
+   skip_before_filter :require_no_authentication
```
Comment (manazirahsan) [inline app/controllers/registrations_controller.rb]: Convention

**Comment by manazirahsan** [comment]
Ok

File: app/controllers/registrations_controller.rb [hunk 6]
```diff
@@ -1,9 +1,31 @@
...
+    :role
+  end
+
+  def account_update_params
+  params.require(:user).permit :first_name, :last_name, :email, :password,
+    :password_confirmation, :current_password
+  end
+
+  def check_supervisor
+  if user_signed_in? && !current_user.supervisor?
```
Comment (manazirahsan) [inline app/controllers/registrations_controller.rb]: Convention

**Comment by manazirahsan** [comment]
ok

**Pull request merged**